walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,18 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Arrays;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    //throw new UnsupportedOperationException("TODO: Implement this method.");
+    if (events.isEmpty()) {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    if (request.getDuration() > (long)TimeRange.WHOLE_DAY.duration()) {
+        return Arrays.asList();
```

Comment:
no change needed here but just as an fyi look into `Collections.emtpyList()` which is immutable. `Arrays.asList()` returns a new `ArrayList()` which means it is mutable. This is not going to be an issue with your current code but one thing to always think about when returning objects is whether allowing mutability would cause any issues.

Reply:
Ohhh ok. Makes sense